include/linux/find_closest.h

```diff
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Find closest element functions
+ *
+ * Copyright (C) 2025 Alexandru Soponar <asoponar@taladin.ro>
+ * Based on util_macros.h implementation by Alexandru Ardelean <aardelean@baylibre.com>
+ */
+#ifndef _LINUX_FIND_CLOSEST_H_
+#define _LINUX_FIND_CLOSEST_H_
+
+#include <linux/types.h>
+
+/**
+ * find_closest_s32 - locate the closest element in a sorted array
+ * @x: The reference value.
+ * @a: The array in which to look for the closest element. Must be sorted
+ *  in ascending order.
+ * @as: Size of 'a'.
+ *
+ * Returns the index of the element closest to 'x'.
+ * Note: This function is specifically designed for s32 (signed 32-bit) arrays
+ * to properly handle negative and positive values.
+ */
+unsigned int find_closest_s32(s32 x, const s32 *a, unsigned int as);
```

Comment:
the `find_closest_s32()` naming is interesting;
initially, i thought maybe about doing it as just `find_closest()`, but i can see it being useful to have it as `find_closest_s32()`;

no idea what upstream will say;
let's leave it as-is for now;

as context: sometimes future-proofing is discouraged, especially in cases where there is no clear idea if there will ever be a `find_closest_u32`  or `find_closest_s64` version;  in the kernel, things are completely self-contained/isolated, so if there is no clear idea if something will ever be used in the future, it is discouraged and the remark will be "we can change it later if we need to"